src/main/cocoa/CocoaWindow.mm

```diff
@@ -0,0 +1,731 @@
+                return STATUS_OK;
+            }
+
+            status_t CocoaWindow::set_caption(const char *caption)
+            {
+                if (caption == nullptr)
+                    return STATUS_BAD_ARGUMENTS;
+
+                LSPString tmp;
+                return (tmp.set_utf8(caption)) ? set_caption(&tmp) : STATUS_NO_MEM;
+            }
+
+            status_t CocoaWindow::get_caption(LSPString *text)
+            {
+                if (text == nullptr)
+                    return STATUS_BAD_ARGUMENTS;
+                if (pCocoaWindow == nullptr)
+                    return STATUS_BAD_STATE;
+
+                NSString *title = [pCocoaWindow title];
+                const char *utf8 = [title UTF8String];
+                return (text->set_utf8(utf8)) ? STATUS_OK : STATUS_NO_MEM;
+            }
+
+            status_t CocoaWindow::get_caption(char *text, size_t len)
+            {
+                if (text == nullptr)
+                    return STATUS_BAD_ARGUMENTS;
+                if (len < 1)
+                    return STATUS_TOO_BIG;
+
+                LSPString tmp;
+                status_t res = get_caption(&tmp);
+                if (res != STATUS_OK)
+                    return res;
+
+                const char *utf8 = tmp.get_utf8();
+                size_t count = strlen(utf8) + 1;
+                if (len < count)
+                    return STATUS_TOO_BIG;
+
+                memcpy(text, utf8, count);
+                return STATUS_OK;
+            }
+
+            //TODO: Translate all Window Border Styles!
+            static NSWindowStyleMask get_ns_style(border_style_t style) {
+                using namespace lsp::ws;
+                switch (style) {
+                    case BS_DIALOG:         return NSWindowStyleMaskTitled;
+                    case BS_SINGLE:         return NSWindowStyleMaskTitled | NSWindowStyleMaskClosable | NSWindowStyleMaskMiniaturizable;
+                    case BS_NONE:           return NSWindowStyleMaskBorderless;
+                    /*
+                    case BS_POPUP:
+                    case BS_COMBO:
+                    */
+                    case BS_DROPDOWN:       return NSWindowStyleMaskBorderless;
+                    case BS_SIZEABLE:       return NSWindowStyleMaskTitled | NSWindowStyleMaskResizable |
+                                                NSWindowStyleMaskClosable | NSWindowStyleMaskMiniaturizable;
+                    default:                return NSWindowStyleMaskTitled;
+                }
+            }
+
+            //TODO: Translate all Window Border Styles!
+            status_t CocoaWindow::get_border_style(border_style_t *style) {
+                using namespace lsp::ws;
+
+                if (style == NULL)
+                    return STATUS_BAD_ARGUMENTS;
+
+                if (pCocoaWindow == NULL)
+                    return STATUS_BAD_STATE;
+
+                NSWindowStyleMask windowStyle = [pCocoaWindow styleMask];
```

Comment:
Maybe it is better just to remember border style inside of the `CocoaWindow` object and return it? I suppose no one except the calling code will force the border style to change.

Reply:
You a right, I will implemet this.